tests/scripts/all.sh

```diff
@@ -4005,14 +4015,45 @@ test_build_opt () {
     done
 }
 
-component_test_clang_opt () {
+# For FreeBSD we invoke the function by name so this condition is added
+# to disable the existing test_clang_opt function for linux.
+if [[ $(uname) != "Linux" ]]; then
+    component_test_clang_opt () {
+        scripts/config.py full
+        test_build_opt 'full config' clang -O0 -Os -O2
+    }
+fi
+
+component_test_clang_latest_opt () {
+    scripts/config.py full
+    test_build_opt 'full config' "$CLANG_LATEST" -O0 -Os -O2
+}
+support_test_clang_latest_opt () {
+    type "$CLANG_LATEST" >/dev/null 2>/dev/null
+}
+
+component_test_clang_earliest_opt () {
     scripts/config.py full
-    test_build_opt 'full config' clang -O0 -Os -O2
+    test_build_opt 'full config' "$CLANG_EARLIEST" -O0
```

Comment:
Why `-O0`? That's the fastest build, but `-O` or -O2` is faster when doing build and test. Also, `-O0` omits some warnings.

Reply:
My bad, I looked into the changes thoroughly and this was supposed to be `-O`. :( 
Its also mentioned in #7891. 
> For the earliest compilers, just do a -O build.

Shall I create a new PR for this change?

Reply:
I'll fix it in https://github.com/Mbed-TLS/mbedtls/pull/9286 which is where I noticed it.

Reply:
Thank you.